editor/src/clj/editor/model_scene.clj

```diff
@@ -440,6 +439,33 @@
         :children (mapv #(augment-model-scene % old-node-id new-node-id new-node-outline-key material-name->material-scene-info)
                         model-scenes)))))
 
+(defn make-material-name->material-scene-info
+  "Given some material-scene-infos, return a material-name->material-scene-info
+  fn suitable for use with the augment-scene function."
+  [material-scene-infos]
+  (let [;; When augmenting the scene, we only want to use material-scene-infos
+        ;; that are fully formed, and ignore the others.
+        usable-material-scene-infos
+        (filterv
+          (fn [material-scene-info]
+            (and (some? (:shader material-scene-info))
+                 (some? (:vertex-space material-scene-info))))
+          material-scene-infos)
+
+        ;; If we have no material associated with the index, we mirror the
+        ;; engine behavior by picking the first one:
+        ;; https://github.com/defold/defold/blob/a265a1714dc892eea285d54eae61d0846b48899d/engine/gamesys/src/gamesys/resources/res_model.cpp#L234-L238
+        fallback-material-scene-info
+        (first usable-material-scene-infos)
+
+        usable-material-scene-infos-by-material-name
+        (->> usable-material-scene-infos
+             (coll/pair-map-by :name)
+             (coll/not-empty))]
```

Comment:
`not-empty` because we don't need a bunch of unique empty map objects hanging around.